File: chat-room-common/src/main/java/io/github/wj0410/chatroom/common/message/SyncOnlineMessage.java
package io.github.wj0410.chatroom.common.message;

import io.github.wj0410.chatroom.common.model.ClientModel;
import lombok.Data;

import java.util.LinkedHashSet;

/**
 * @author wangjie
 * @date 2023/10/26
 */
@Data
public class SyncOnlineMessage {
    private LinkedHashSet<ClientModel> clientOnlineList;

    @Override
    public String toString() {
        return "SyncOnlineMessage{" +
                "clientOnlineList=" + clientOnlineList +
                '}';
    }
}

File: chat_vue/src/model/SyncOnlineMessage.ts
import {Message} from "./Message";
import {MessageType} from "../constant/Enum";
import ClientModel from "./ClientModel.ts";

export class SyncOnlineMessage {

    public clientOnlineList: Array<ClientModel>


    constructor(data) {
        this.clientOnlineList = data.clientOnlineList;
    }

    buildMessage() {
        let obj = {
            type: MessageType.SYNC_ONLINE,
            data: this
        }
        return new Message(obj);
    }
}
